base/src/snark_work.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+//! Types related to the Transaction Snark Work
+#![allow(missing_docs)]
+
+use crate::types::*;
+use mina_crypto::hash::*;
+use proof_systems::mina_signer::CompressedPubKey;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct TransactionSnarkWork {
+    pub fee: Amount,
+    pub proofs: OneORTwo,
+    pub prover: CompressedPubKey,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
```

Comment:
You can drop the `Serialize`, `Deserialize` derives and also the polyvar tag. These are only required for serialization types and now we are in Rust land they are just enums

Reply:
fixed